### PR TITLE
EAR-1559-header component

### PR DESCRIPTION
**Pull request by shelleytruong**
> ### BEFORE MAKING YOUR PR
>
> Please ensure:
>
> - There are no linting errors, all tests must pass
> - PR is named after JIRA ticket number e.g. EAR-###
> - **Accesibility** checks are completed:
>   - Elements have discernible and consistent focus states
>   - Elements can be navigated to and used by just a keyboard
>   - Elements and text can be read out by a screen reader, and the descriptions are understandable
> - Your feature / bug fix works across **GCP** and **AWS** (where appropriate)
>   - Are modifications to eq-publisher-v3 required?
>   - Are modifications to the Firestore data source required?

---

### What is the context of this PR?

>Header component based on ONS design system - internal, internal with description, external and external with description variants used.
> 
### How to review

> Add to the list below as appropriate, including screenshots when necessary

- Import the **Capability examples** questionnaire from pre-prod Author into your local environment and:
  - ensure it can be opened in Author;
  - then, ensure it can be viewed in Runner by pressing the **view survey** button
- Does this require a migration? We need one if existing JSON schema properties change

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process

### CODE REVIEW

**Review by ashtuck — DISMISSED** [review]

File: eq-author/src/components-themed/Header/index.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  ${({ withDescription }) =>
+    withDescription &&
+    `font-size: 2rem;
+  `}
+`;
+
+const HeaderMain = styled.div`
+  background-color: ${({ theme }) => theme.colors.headerMain};
+  padding: 0.56rem 0;
+  padding-left: 18px;
```
Comment (martyncolmer) [inline eq-author/src/components-themed/Header/index.js:49]: Can we make this 1 rem and have the padding on the right as well.

**Review by martyncolmer — CHANGES_REQUESTED** [review]
Noted a few things. 
Some additional things we will want to discuss. 
Theres no option to have side content such as a button.
Theres no option to have list options on the bottom of the header.

File: eq-author/src/components-themed/Header/index.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+const Header = ({
+  variant,
+  children,
+  headerDescription,
+  logo,
+  exampleList,
+}) => {
+  return (
+    <>
+      <HeaderTop variant={variant} exampleList={exampleList}>
```
Comment (martyncolmer) [inline eq-author/src/components-themed/Header/index.js:77]: Don't think you need to pass in example list here as not used by css

File: eq-author/src/components-themed/Header/index.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  padding-right: 1rem;
+  height: auto;
+`;
+
+const Header = ({
+  variant,
+  children,
+  headerDescription,
+  logo,
+  exampleList,
```
Comment (martyncolmer) [inline eq-author/src/components-themed/Header/index.js:73]: need a better name like headerTopContent

File: eq-author/src/components-themed/Header/index.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    fill: ${({ theme, variant }) =>
+      variant === "Internal"
+        ? theme.colors.internalOnsLogo
+        : theme.colors.externalOnsLogo};
+  }
+  ul {
+    li {
+      display: inline-block;
+      margin: 0 0 0 1rem;
+      font-size: 1rem;
```
Comment (martyncolmer) [inline eq-author/src/components-themed/Header/index.js:28]: add this to move the underline lower
text-underline-position: under;

**Review by martyncolmer — CHANGES_REQUESTED** [review]

**Review by ashtuck — APPROVED** [review]

**Review by martyncolmer — APPROVED** [review]

**Review by ashtuck — APPROVED** [review]

**Pull request merged**